File: tokamak-server/src/main/java/fm/pattern/tokamak/server/service/PasswordPolicyService.java
package fm.pattern.tokamak.server.service;

import java.util.List;

import fm.pattern.tokamak.server.model.PasswordPolicy;
import fm.pattern.valex.Result;
import fm.pattern.valex.annotations.Create;
import fm.pattern.valex.annotations.Delete;
import fm.pattern.valex.annotations.Update;

public interface PasswordPolicyService {

	Result<PasswordPolicy> create(@Create PasswordPolicy policy);

	Result<PasswordPolicy> update(@Update PasswordPolicy policy);

	Result<PasswordPolicy> delete(@Delete PasswordPolicy policy);

	Result<PasswordPolicy> findById(String id);
	
	Result<PasswordPolicy> findByName(String name);

	Result<List<PasswordPolicy>> list();
	
}

File: tokamak-console/app/src/services/PasswordPolicyService.js
import RestService from "./RestService.js"

var PasswordPolicy = {
        
    list() {
        return RestService.list("/v1/policies");
    },        
        
    update(passwordPolicy) {
        return RestService.put("/v1/policies/" + passwordPolicy.id, passwordPolicy);
    },

    findByName(name) {
        return RestService.get("/v1/policies/name/" + name);
    },
    
}

export default PasswordPolicy;